internal/immutable/immutable.go

```diff
@@ -0,0 +1,22 @@
+package immutable
+
+// Immutable represents immutable chains
```

Comment:
hey @xiam can you include some more info in the comments here what this interface is typically used for?